ivy/functional/backends/numpy/activations.py

```diff
@@ -25,10 +25,14 @@ def leaky_relu(
 
 @_scalar_output_to_0d_array
 def gelu(
-    x, /, *, approximate: bool = False, out: Optional[np.ndarray] = None
+    x : np.ndarray,
+    /,
+    *,
+    approximate: bool = False,
+    out: Optional[np.ndarray] = None
 ) -> np.ndarray:
     if approximate:
-        ret = 0.5 * x * (1 + np.tanh(x * 0.7978845608 * (1 + 0.044715 * x * x)))
+        ret = 0.5 * x * (1 + np.tanh(0.7978845608 * (x + 0.044715 * x * x * x)))
     else:
         ret = 0.5 * x * (1 + ivy.erf(x / np.sqrt(2)))
```

Comment:
When writing a backend implementation for a function as a composition, only the backend-specific functions should be used. Here you're using an `ivy` function (**ivy.erf()**) when implementing `gelu` for numpy backend. Also, you should use `np.astype()` instead of `ivy.astype()`

Reply:
@YushaArif99 it is just , numpy has no erf method.

Reply:
You could, however, this would create a dependency on the scipy module which is not something we want. Alternatively, I'd suggest creating your own NumPy-compatible version of the `erf` function. The scipy documentation defines the function as being a purely mathematical one ` 2/sqrt(pi)*integral(exp(-t**2), t=0..z)` and so implementing it should be straightforward. I think it would be better to implement this function for the NumPy backend itself.
One question that comes to mind though is that if there is an `ivy.erf()` function, isn't there also a NumPy backend implementation of this function as well? If so, you should directly use that. If it's not there for some reason, then it's best to implement it in the NumPy backend so that `ivy.erf()` works when using a NumPy backend. 
Hope that makes sense 🙂

Reply:
@YushaArif99  Unfortunately for some reasons there is no erf in the NumPy. I understood you , but where exactly should I implement the ivy.erf() in the NumPy backend. What I mean should I just make it at the top of the activations.py file or in another file?

Reply:
Well I looked into the master branch of Ivy and noticed that the `erf` function has already been implemented for the NumPy backend. In this case, I think it's okay to use `ivy.erf()` since there is an existing NumPy implementation of it in Ivy